src/mlpack/methods/preprocess/preprocess_split_main.cpp

```diff
@@ -76,10 +76,13 @@ static void mlpackMain()
   // Parse command line options.
   const double testRatio = CLI::GetParam<double>("test_ratio");
 
-  if (CLI::GetParam<int>("seed") == 0)
-    mlpack::math::RandomSeed(std::time(NULL));
-  else
-    mlpack::math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+  if (BINDING_TYPE != BINDING_TYPE_TEST)
```

Comment:
Two things:

1. By using `if` and not `#if`, this check gets compiled into the code (unless the code is properly optimized), so it is better to use the preprocessor here.  In general I would say if you are dealing with preprocessor macros, you should use the logical functionality provided by the preprocessor.

2. If we perform the check here, inside the bindings, we have to modify every single binding that has a `seed` option.  Wouldn't it be better to make changes to the `RandomSeed()` function directly?